src/site/content/en/blog/hands-on-portals/index.md

```diff
@@ -0,0 +1,196 @@
+    z-index: 10000;
+  }
+  .portal-reveal {
+    transform: scale(1.0);
+    bottom: 0px;
+    left: 0px;
+  }
+  .fade-in {
+    opacity: 1.0;
+  }
+`;
+const portal = document.createElement('portal');
+// Let’s navigate into the WICG Portals spec page
+portal.src = 'https://wicg.github.io/portals/';
+portal.addEventListener('click', evt => {
+  // Animate the portal once user interacts
+  portal.classList.add('portal-reveal');
+});
+portal.addEventListener('transitionend', evt => {
+  if (evt.propertyName == 'bottom') {
+    // Activate the portal once the transition has completed
+    portal.activate();
+  }
+});
+document.body.append(style, portal);
+
+// Waiting for the page to load.
+// using setTimeout is a suboptimal way and it’s best to fade-in
+// when receiving a load complete message from the portal via postMessage
+setTimeout(_ => portal.classList.add('fade-in'), 2000);
+```
+
+It is also easy to do feature detection to progressively enhance a website using Portals.
+
+```javascript
+if ('HTMLPortalElement' in window) {
+  // If this is a platform that has Portals...
+  const portal = document.createElement('portal');
+  ...
+}
+```
+
+If you want to quickly experience what Portals feels like, try using [uskay-portals-demo.glitch.me](https://uskay-portals-demo.glitch.me). 1) Enter a URL you want to preview, 2) the page will then be embedded as a portal element, 3) click on the preview 4) The preview will be activated after an animation.
+
+![A gif of using the glitch demo](glitch.gif)
+
+# Check out the spec
+We are actively discussing the spec in [WICG](https://github.com/WICG/portals/blob/master/explainer.md). You can check out the details [here](https://wicg.github.io/portals/) but overall there are three important features that builds up the element.
+
+ - [The portal element](https://wicg.github.io/portals/#the-portal-element) … The HTML element itself. The API is very simple. It consists of the `src` attribute, the `activate` function and an interface for messaging (`postMessage`). activate has an optional argument to pass data to the portal upon activation.
+ - [The portal host interface](https://wicg.github.io/portals/#the-portalhost-interface) … This adds a `portalHost` object to the window object. This lets you check if the page is embedded as a portal element. It also provides an interface for messaging (`postMessage`) back to the host.
+ - [The PortalActivateEvent interface](https://wicg.github.io/portals/#the-portalactivateevent-interface) … An event being fired when the portal is activated. There is a neat function called `adoptPredecessor` which you can use to retrieve the previous page as a `<portal>  ` element. This allows you to create seamless navigations and composed experiences between two pages.
+
+Let’s look beyond the basic usage pattern. Here is a non-exhaustive list of you can achieve with Portals along with sample code.
+## Customize the style when embedded as a portal element
```

Comment:
If you want you could turn some or all of these into embedded glitches to let folks preview the code snippet. That's totally optional. Only if you think it will help the article. I realize at the end there's a video which kind of ties it all together but I think having things folks can play around with can be fun.

Reply:
Thanks for the suggestion! Unfortunately, these are not so useful to just have them individually running on glitch. I would rather want to keep the snippet as is. And lead them to the WICG demos repo in the end (and the vid).

Reply:
sounds good.